### PR TITLE
Feature: IEO plugin implementation

### PR DESCRIPTION
**Pull request by josadcha**
Feature: add IEO modules

Feature: implement IEO list container

Feature: add IEO countdown ticker

Feature: add IEO profile screen structure

Feature: add saga for fetching IEO item

Feature: implementation of IEO Order component

Feature: modify get IEO saga for fetching with metadata

Feature: connect IEO list to websocket

Feature: add light theme for ieo

Feature: add ieo order exececute success modal

### CODE REVIEW

File: src/plugins/ieo/components/IEOCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    currencies: Currency[];
+}
+
+interface State {
+    countdownValue: string;
+}
+
+const limitDescription = 80;
+
+type Props = CardIEOProps & InjectedIntlProps;
```
Comment (akhlopiachyi) [inline src/plugins/ieo/components/IEOCard/index.tsx:23]: add empty line

File: src/plugins/ieo/containers/IEOListElement/index.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        ieo && this.props.history.push(`/ieo/${ieo.id}`);
+    };
+}
+
+const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = dispatch => ({
+    setCurrentIEO: payload => dispatch(setCurrentIEO(payload)),
+});
+
+// tslint:disable-next-line:no-any
+export const IEOListElement = injectIntl(withRouter(connect(null, mapDispatchToProps)(IEOListContainer) as any));
```
Comment (akhlopiachyi) [inline src/plugins/ieo/containers/IEOListElement/index.tsx:63]: use compose

File: src/plugins/ieo/containers/IEOInfo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    private onSignInClick = () => {
+        this.props.history.push('/signin');
+    };
+
+    private onSignUpClick = () => {
+        this.props.history.push('/signup');
+    };
+}
+
+export const IEOInfo = injectIntl(withRouter(IEOInfoComponent));
```
Comment (akhlopiachyi) [inline src/plugins/ieo/containers/IEOInfo/index.tsx:323]: use compose

File: src/plugins/ieo/containers/OrderIEO/index.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+const mapDispatchToProps = dispatch => ({
+    fetchCurrencies: () => dispatch(currenciesFetch()),
+    executeIEOOrder: payload => dispatch(ieoOrderExecuteFetch(payload)),
+    accountWallets: () => dispatch(walletsFetch()),
+    ieoOrdersPush: payload => dispatch(ieoOrdersPush(payload)),
+});
+
+// tslint:disable-next-line no-any
+const OrderIEO = injectIntl(connect(mapStateToProps, mapDispatchToProps)(OrderIEOContainer as any)) as any;
```
Comment (akhlopiachyi) [inline src/plugins/ieo/containers/OrderIEO/index.tsx:185]: use compose

**Comment by josadcha** [comment]
this PR is the basis for implementation of separate IEO plugin, it should not be merged to master, but instead overridden according to plugin's principle

**Comment by josadcha** [comment]
moved to https://github.com/openware/baseapp-plugin-ieo

**Pull request closed without merging**